core/src/main/java/hudson/model/Label.java

```diff
@@ -389,7 +389,7 @@ public int getTiedJobCount() {
                         for (Item i : parent.getItems()) {
                             if (i instanceof AbstractProject) {
                                 final AbstractProject project = (AbstractProject) i;
-                                if (this.equals(project.getAssignedLabel())) {
+                                if (this.name.equals(project.getAssignedLabelString())) {
```

Comment:
There are no actual `Label` implementations overriding `equals` in core, and plugins cannot fruitfully contribute subclasses; they are all constructed by `LabelExpressionParser`.